### PR TITLE
Fix js tracker changing request for axios + revert error and result in callback + fix test + webpack config

### PR DESCRIPTION
**Pull request by jujusb**

### CODE REVIEW

File: src/js-tracker.js [hunk 1]
```diff
@@ -15,17 +15,29 @@
...
-'use strict';
 
-var request = require('request');
+var axios = require('axios');
 var moment = require('moment');
+var uuidv4 = require('uuid').v4;
+var ScalableBloomFilter = require('bloom-filters').ScalableBloomFilter;
+
+// by default it creates an ideally scalable bloom filter for 8 elements with an error rate of 0.01 and a load factor of 0.5
+var filter = new ScalableBloomFilter();
```
Comment (imartinezortiz) [inline src/js-tracker.js:25]: The bloomfilter it only required in the backend server. It is better not have this dependency in a potential js client.

File: src/js-tracker.js [hunk 2]
```diff
@@ -49,8 +61,9 @@ function TrackerAsset() {
     this.active = false;
 
     this.session = 0;
-    this.actor = '{}';
+    this.actor = new TrackerEvent.TraceActor("undefined", "undefined", "undefined");
```
Comment (imartinezortiz) [inline src/js-tracker.js:64]: Not a very good programming practice. Better allow null and check that is not null when the this.actor is required.

File: src/js-tracker.js [hunk 3]
```diff
@@ -95,53 +108,49 @@ function TrackerAsset() {
...
+                }
+                callback(null, data);
+            });
     };
 
     this.LoginBeaconing = function(accessToken, callback) {
         this.generateURL();
 
         var tracker = this;
         this.HttpRequest(this.url + 'login/beaconing', 'post', {'Content-Type': 'application/json' }, JSON.stringify({accessToken: accessToken}),
```
Comment (imartinezortiz) [inline src/js-tracker.js:130]: Open an issue to track this URLs down. Any reference to beaconing / rage should be removed eventually. They are just dead code not required anymore.

**Review by imartinezortiz — CHANGES_REQUESTED** [review]

File: src/js-tracker.js (hunk 2 above)
Comment (jujusb) [reply src/js-tracker.js]: fixed

File: src/js-tracker.js (hunk 1 above)
Comment (jujusb) [reply src/js-tracker.js]: fixed

File: src/js-tracker.js (hunk 3 above)
Comment (jujusb) [reply src/js-tracker.js:122]: issue opened

**Review by imartinezortiz — APPROVED** [review]

File: src/js-tracker.js [hunk 4]
```diff
@@ -757,7 +757,7 @@ function TrackerEvent (tracker) {
     this.ToXapi = function() {
         var t = {
             id : this.Id,
-            actor: this.getActor().ToXapi(),
+            actor: this.getActor() ? this.getActor().ToXapi() : null,
```
Comment (imartinezortiz) [inline src/js-tracker.js:760]: this.getActor()?.ToXapi()

**Review by imartinezortiz — APPROVED** [review]

File: src/js-tracker.js (hunk 4 above)
Comment (jujusb) [reply src/js-tracker.js:760]: Removed

File: src/js-tracker.js [hunk 5]
```diff
@@ -198,11 +198,10 @@ function TrackerAsset() {
                 }
 
                 tracker.auth = data.authToken;
-                if(data.actor) {
-                    tracker.actor = new TrackerEvent.TraceActor(data.actor.name, data.actor.account.name, data.actor.account.homePage);
-                } else {
-                    tracker.actor = new TrackerEvent.TraceActor("undefined", "undefined", "undefined");
-                }
+                actor_name = data.actor.name == null ? "undefined" : data.actor.name;
+                account_name = data.actor.account.name == null ? "undefined" : data.actor.account.name;
```
Comment (imartinezortiz) [inline src/js-tracker.js:202]: actor_name and account_name must never be null neither undefined, at least should be a token "XYZ..". If not available should raise an error.

**Review by imartinezortiz — CHANGES_REQUESTED** [review]

File: src/js-tracker.js (hunk 5 above)
Comment (jujusb) [reply src/js-tracker.js]: fix

Comment (imartinezortiz) [reply src/js-tracker.js]: Using "XYZ.." was just an example. The best option is to raise an error, but if not feasible, define a "token" that is understandable.

Comment (jujusb) [reply src/js-tracker.js]: raise a message now

**Review by imartinezortiz — APPROVED** [review]

**Pull request merged**